6 june 2026 meetup (#235)
nairobilug/nairobilug.or.ke: content/meetup-june-2026.md

Title: June 2026 Meetup.
Date: 2026-06-6 15:00
Category: Meetups
Tags: TCP, opensource, Kuernetes, mku-techinnovation meetup
Slug: meetup-june-2026
Author: Steve Nyasimi 
Summary: Highlights from the June, 2026 meetup.

### Introduction 
Hello, I hope you're okay. 

> You know, NaiLug has always felt like home. The community is full of kind and awesome guys. From the first meet up this year, which was on the 14<sup>th</sup> of February 2026 at KFC till date. After all, it was <del>Valentine's day</del> I love Free Software day, You check out how it went [here](https://nairobilug.org/2026/02/meetup-february-2026.html).   

Reviewer: ```suggestion
NaiLug has always felt like home. The community is full of kind and awesome guys. From the first meet up this year, which was on the 14<sup>th</sup> of February 2026 at KFC till date. After all, it was <del>Valentine's day</del> I love Free Software day, You check out how it went [here](https://nairobilug.org/2026/02/meetup-february-2026.html).   
```
I think this sounds better.

> But this time, I wanted to give back, something to show my appreciation. Not in a big way, but in a way i could help the community grow and keep the meet ups going. Well, since i was a student, i wanted to give something other than talks.* I had done one during a previous meet, it was about bash scripting, you can find the video [here](https://youtu.be/8XLRnMF_wY8)*. I wanted to do more, I gathered enough courage and volunteered to host the next meet up, i had a small plan in mind and that would be to host the next event in Mounk Kenya University, Nairobi campus. This would help the comminity and at the same time introduce my peers to open source and the amazing community. 

Reviewer: ```suggestion
This time, I wanted to give back.  To show my appreciation. Not in a big way, but in a way I could help the community grow and keep the meet-ups going. I was still a student and I thought of a way of contributing to the community other than by giving a talk.* I had already given a talk about bash-scripting. The recording can be found [here](https://youtu.be/8XLRnMF_wY8)*. In my quest to do more, I gathered enough courage and volunteered to host the next meet up,. The  plan in mind was to  to host the next event in Mount Kenya University, Nairobi campus. This would help the community with a free venue and at the same time introduce my peers to open source and the amazing community at large. 
```

> At first i was a bit skeptikal, regardless I took the initiative and engaged our school coordinator and asked if we could host the meetup in MKU. She agreed and we got to planning. I was happy, and remembered what my big bro, Mr.Weekends always tells me.. 

Reviewer 2: Your bro is Mr. Weekends? 😆

Author: Yes 😂😂

Reviewer: 😆 😆

> At first i was a bit skeptikal, regardless I took the initiative and engaged our school coordinator and asked if we could host the meetup in MKU. She agreed and we got to planning. I was happy, and remembered what my big bro, Mr.Weekends always tells me.. 

Reviewer: ```suggestion
At first, I  was a bit skeptical, regardless I took the initiative and engaged our school coordinator, Dr. Kiboori. I asked if we could host the meetup at MKU. After a bit of back and forth in communication,  she agreed and we got to planning. I was happy, and remembered what my big bro, Mr.Weekends always tells me.. 
```

> *If you do not ask, the answer will always be no!*

This blog will be about Nailug meetup that happened on the 6<sup>th</sup> of june 2026.

> NaiLug is full of amazing people.

Reviewer: ```suggestion
This blog is about the meetup that happened on the 6<sup>th</sup> of June 2026.
NaiLug is full of amazing people.
```

### What went down
Saturday afternoon, 15:00hrs, seats arranged, projector ready and mics tested. A simple set up for the nailug event. 

> The first person came in, my heart was pumping full of exitement and anxiety at the same time, I didn't know what to expect as it was my first time organizing and hosting an event at our school. But all the chaos subsided when I saw my friends and members of the community come in "ah, finally some familiar faces, phew." i said to my self. 

Reviewer: ```suggestion
Saturday afternoon, 15:00hrs, seats arranged, projector ready and mics tested. A simple set up for the event. 
The first person came in, my heart was pumping full of excitement and anxiety at the same time, I didn't know what to expect. It was my first time organizing and hosting an event at our school! But all the fear and doubt subsided when I saw my friends and members of the community come in "ah, finally some familiar faces, phew.", I said to my self. 
```

> I also had the privilege of being both host and speaker of the event, the event was facilitated by Cornelius Emase, our MC.

Reviewer: ```suggestion
I  had the privilege of being both host and speaker of the event. The event was facilitated by Cornelius Emase, our MC.
```

> We had a line up of awesome talks and presentations. The main ones were

Reviewer: ```suggestion
We had a line up of awesome talks and presentations. The main ones were:
```

- Building HTTP from scratch using rust by Daniel Muringe

> - Kubernetes Talk by Steve Nyasimi

Reviewer: ```suggestion
- Building HTTP from scratch using Rust by [Daniel Muringe](https://www.linkedin.com/in/danmuringe/)
- Kubernetes Talk by [Steve Nyasimi](https://www.linkedin.com/in/steve-solomon-39ab72248/)
```

> Both were very engaging sessions with alot of knowledge being passed around, honestly this is one of the reasons i love this community. People get together, share ideas and knowledge and network. The love that's there, unmatched!

Reviewer: ```suggestion

Both were intended to be very engaging sessions with a lot of knowledge being passed around. Honestly this is one of the reasons I love this community. People get together, share ideas and knowledge and network. The love that's there, unmatched!
```

#### Talk 1: Building HTTP from sctatch by Daniel Muringe

> <iframe title="Building HTTP from scratch using Rust" width="560" height="315" src="https://www.youtube.com/embed/cQoA3-6KjXo?si=Vqwdj9obwcPwZgK4" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

Reviewer: Where did you get this? @Snyasimi

Author: Youtube's embed

Reviewer 2: Interesting. where did you get it? 

Videos are put on the website like this https://youtu.be/xb9fsBoWpEs?si=9-wswZqn1GSGXj9n as we have a plugin that allow it.

Have a look at this entry: https://raw.githubusercontent.com/nairobilug/nairobilug.or.ke/refs/heads/master/content/meetup-may-2026-uhuru-uhuru-uhuru-uhuru-tidesdb.md

Something like https://github.com/nairobilug/nairobilug.or.ke/blame/45de892e80ba63e2a1871c32adeeb5c54743e9fb/content/meetup-may-2026-uhuru-uhuru-uhuru-uhuru-tidesdb.md#L36

Author: When you want to share a video from youtube, it gives you the option to embed an iframe.
Will replace that

Reviewer 2: No worries. I tested it out and it works great.

Reviewer 2: Nothing to worry about there

> Daniel Muringe opened the event with his presentation on how you can implement HTTP from scratch, this was cool as _" There's no better way to learn how something works other than building it"_, Daniel. 

Reviewer: ```suggestion
Daniel Muringe opened the event with his presentation on how you can implement HTTP from scratch. This was cool as _" There's no better way to learn how something works other than building it"_, Daniel. 
```

> He took us through the OSI model and a brief history on how it came to be. He also shared alot on the different versions of the HTTP protocal, TCP and UDP and how HTTP builds on TCP. He made sure we had a reliable connection with him as he was explaining.

Reviewer: ```suggestion
He took us through the OSI model and a brief history of how it came to be. He also shared a lot on the different versions of the HTTP protocols, TCP and UDP and how HTTP builds on TCP. He made sure we had a reliable connection with him as he was explaining.
```

He ended his talk with a short fun session of Q and A where the audience got to ask questions and he'd answer them.

##### Key take aways
- TCP is a reliable connection compared to UDP.
- There are different types of HTTP and each version builds on it's predecessor.
- Rust's memory safety features can help you with memory bugs.

#### Talk 2: An introduction to kubernetes
 
 <iframe title="an introduction to kubernetes" width="560" height="315" src="https://www.youtube.com/embed/cQoA3-6KjXo?si=Vqwdj9obwcPwZgK4" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture; web-share" referrerpolicy="strict-origin-when-cross-origin" allowfullscreen></iframe>

> Our next speaker was Steve Nyasimi, he went over kubernetes. Gave us an introduction to what Kubernetes is, the architecture and a brief history of kubernetes. From describing the smallest unit in kubernetes to how kubernetes handles traffic and how you can use the kubernetes deployments to ensure high availability of workloads.  

Reviewer: ```suggestion
I was the next speaker. In my  Kubernetes talk, I gave an introduction to what Kubernetes is, its  architecture and a brief history about it. I described the smallest unit in Kubernetes - the pod - , how Kubernetes handles traffic and how it can be used to ensure high availability of apps though horizontal scaling.
```

> We went through containers, docker, pods, deployments and services and how they all work together to handle jobs and workloads. The audience and the speaker also got to exchange knowledge and information. 

Reviewer: ```suggestion
I went through containers, docker, pods, deployments and services and how they all work together to handle jobs and workloads. The audience and I also got to exchange knowledge and information. 
```

> Since we were talking about handling alot of workloads, we also went through some scaling concepts and how kubernetes scales, he went all _out_ honestly. We also went through the concepts of high availability, fault tolerance and orchestration.  

Reviewer: ```suggestion
Since I was talking about how Kubernetes handles  massive workloads.  I went through some scaling concepts and how kubernetes scales. I went all _out_ honestly. I also briefly explained the concepts of high availability, fault tolerance and orchestration.  
```

These concepts were covered through the explanation of kubernetes objects like deployments, replications and pods.

#### Key take aways
- Kubernetes ensures high availability
- K8s supports horizontal scaling to deliver load balancing and fault tolerance.
- Kuberenetes architecture
- Horizontal and vertial scaling

### Networking, sharing and engagement.
We finished the talks and started the networking session over juice, biscuits and snacks. The room was full of energy, people networking and socializing. As i was standing in the room enjoying the moment, something crossed my mind...

> *If you want something to happen, make it happen*

You know what, I challenge you, if you can, give back to your community or anywhere you feel that you should.

> It was a cool meet up, i hope to see you guys on the next one. _Cheers!!_

Reviewer: ```suggestion
It was a cool meet up, I hope to see you guys on the next one. _Cheers!!_
```

![Nailug group photo]("https://drive.google.com/file/d/1WkynwTSBeU1jKKed30GeFSH4I3aVX3gK/view?usp=drive_link")
* NaiLug Group session photo

### Acknowledgements
First, I thank the MKU faternity for allowing us and provideing the space to host Nailug, some refreshments and the warm reception. I also thank Mr.Paul, he's been encouraging me to do talks and pushing me to the limit. Faith and Cornelius for their encouragement and warm welcome in the comminity and Benson for his guidance as well. 

I also thank you, the NaiLug community, for always showing up, supporting, and growing the community, because if you guys aren't there, then theres no NaiLug. 

### July Meet up

> According to tradition, we usually meet on the first saturday of every month. Venue will be communicated in due time. I cannot wait to see you all.

Reviewer: ```suggestion
According to tradition, we usually meet on the first Saturday of every month. The venue will be Mount Kenya University, Nairobi Campus . I'm looking forward to seeing you  all on the 4th of July 2026.
```
